webapp/channels/src/components/suggestion/suggestion_list.tsx

```diff
@@ -150,22 +164,23 @@ export default class SuggestionList extends React.PureComponent {
                 return;
             }
 
-            const itemTop = item.offsetTop - this.getComputedCssProperty(item, 'marginTop');
-            const itemBottomMargin = this.getComputedCssProperty(item, 'marginBottom') + this.getComputedCssProperty(item, 'paddingBottom');
-            const itemBottom = item.offsetTop + this.getComputedCssProperty(item, 'height') + itemBottomMargin;
-
-            if (itemTop - contentTopPadding < contentTop) {
-                // the item is off the top of the visible space
-                content.scrollTop = itemTop - contentTopPadding;
-            } else if (itemBottom + contentTopPadding + contentBottomPadding > contentTop + visibleContentHeight) {
-                // the item has gone off the bottom of the visible space
-                content.scrollTop = (itemBottom - visibleContentHeight) + contentTopPadding + contentBottomPadding;
+            if (item instanceof HTMLElement) {
+                const itemTop = item.offsetTop - this.getComputedCssProperty(item, 'marginTop');
+                const itemBottomMargin = this.getComputedCssProperty(item, 'marginBottom') + this.getComputedCssProperty(item, 'paddingBottom');
+                const itemBottom = item.offsetTop + this.getComputedCssProperty(item, 'height') + itemBottomMargin;
+                if (itemTop - contentTopPadding < contentTop) {
+                    // the item is off the top of the visible space
+                    content.scrollTop = itemTop - contentTopPadding;
+                } else if (itemBottom + contentTopPadding + contentBottomPadding > contentTop + visibleContentHeight) {
+                    // the item has gone off the bottom of the visible space
+                    content.scrollTop = (itemBottom - visibleContentHeight) + contentTopPadding + contentBottomPadding;
+                }
             }
         }
     };
 
-    getComputedCssProperty(element, property) {
-        return parseInt(getComputedStyle(element)[property], 10);
+    getComputedCssProperty(element: Element, property: string) {
+        return parseInt(getComputedStyle(element)[property as keyof CSSStyleDeclaration] as string, 10);
```

Comment:
Could you avoid this double casting by typing `property` as `keyof CSSStyleDeclaration` above? If it's possible for it to be undefined or null, the `as string` should be fine then since I know I've caused issues trying to add `|| '0'` in places like this before

Reply:
I tried without the double typecasting as `keyof CSSStyleDeclaration]` but the TypeScript transpiler keeps complaining with the following error: `Element implicitly has an 'any' type because index expression is not of type 'number` on [property].

However, this works: `return parseInt(getComputedStyle(element).getPropertyValue(property), 10);`